pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -307,10 +307,20 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusters(cb *ClusterBuilder, i
 		}
 
 		// For each workload port, we will construct a cluster
-		for _, instances := range clustersToBuild {
-			instance := instances[0]
-			localCluster := cb.buildInboundClusterForPortOrUDS(cb.proxy, int(instance.Endpoint.EndpointPort), actualLocalHost, instance, instances)
-			clusters = cp.conditionallyAppend(clusters, localCluster)
+		if !util.IsIstioVersionGE18(cb.proxy) {
+			// For 1.7 clusters, we used an old cluster format
+			for _, instances := range clustersToBuild {
+				for _, instance := range instances {
```

Comment:
I donot quite understand how this works? 

are there multi services selected the same pod?

Reply:
yes, that is what this change is needed for